var/spack/repos/builtin/packages/py-bluepyemodel/package.py

```diff
@@ -22,7 +22,7 @@ class PyBluepyemodel(PythonPackage):
     depends_on('py-bluepyopt', type='run')
     depends_on('py-bluepyparallel@0.0.3:', type='run')
     depends_on('py-click@7.0:', type='run')
-    depends_on('py-efel@3.0.80:', type='run')
+    depends_on('py-efel@4.0.4:', type='run')
```

Comment:
Why this? Did the current version suddenly require the new eFEL?

If you just bump the version of eFEL, `py-bluepyemodel` will be rebuilt with the newest version already.

Reply:
To use the last version, should I just write ?
`depends_on('py-efel:', type='run')`

Reply:
We basically force Spack to re-evaluate all dependency graphs at every deployment. So by default, it should use the latest version that the dependency graph allows it (other software may restrict the version range)

```
depends_on('py-efel@3.0.80:', type='run') 
```
basically says "use 3.0.80" or newer. If you now restrict it hard to "4.0.4 or greater", if we want to go back and test the interaction between older versions of `py-efel`, we will have to edit the package.

The more restrictive we are with the versions, the higher the likelihood that we will have version conflicts at some point in time.

Reply:
Thanks! I'm not opposed to absolute minimums, but would prefer to keep definitions more on the loose side.

If there's a version of eFEL that should be retired, it would be best to remove it from the package itself (or comment it out stating a reason)